File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        res = 0

        l = 0
        r = len(heights)-1

        while l < r:
            if heights[l] > heights[r]:
                test = heights[r] * (r - l)
                if test > res:
                    res = test

                r -= 1

            else:
                test = heights[l] * (r - l)
                if test > res:
                    res = test

                l += 1

        return res

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int res = 0;
        int l = 0;
        unordered_map<char, int> letMap;
        int maxl = 0;

        for (int r = 0; r < s.size(); r++){
            letMap[s[r]]++;

            if(maxl < letMap[s[r]]){
                maxl = letMap[s[r]];
            }else if(maxl < letMap[s[l]]){
                maxl = letMap[s[l]];
            }

            while ((r-l+1) - maxl > k){
                letMap[s[l]]--;
                l++;
            }

            if ((r-l+1) > res){
                res = r-l+1;
            }
        }

        return res;
    }
};
